data_ingestion/requirements.txt

```diff
@@ -0,0 +1,3 @@
+google-cloud-core==1.4.1
+google-cloud-storage==1.31.0
+requests==2.24.0
```

Comment:
I generated it with pip freeze but then only kept the direct dependencies. I don't think this is the best approach (or even necessarily a good one) but I didn't want to get over-prescriptive with potentially system-dependent dependencies. I just updated it using pip-compile so now we have a requirements-in.txt with the direct deps and a requirements.txt with the indirect ones plus where they come from. We may have to generate separate ones for different OSs if they diverge though. WDYT?

Reply:
I'm not familiar with pip-compile. Is that from here: https://github.com/jazzband/pip-tools ?

I thought that if we used virtual environments then it would avoid system-specific dependencies. Rasmi suggested that the ideal end-state would be a single docker compose command that clones the repo, sets up the virtual environment, and installs dependencies from a requirements file.

This looks like a reasonable way to do things for now. In the spirit of prototyping let's go with it and see how it works.

Reply:
yep, that's the one! IIUC venv (which is all I'm using rn) creates a lightweight environment for package isolation but is still platform dependent. If we went with Docker then we wouldn't need to worry about OS.